website/integrations: nextcloud: add SSE warning (#11976)
goauthentik/authentik: website/integrations/services/nextcloud/index.md

> -- https://en.wikipedia.org/wiki/Nextcloud

:::caution

> If you require [Server Side Encryption](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/encryption_configuration.html), you MUST use LDAP! OpenID and SAML will cause **irrevocable data loss**. Nextcoud Server-Side Encryption requires access to the user's cleartext password, which is not available through OpenID or SAML.

Reviewer: ```suggestion
If you require [Server Side Encryption](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/encryption_configuration.html), you must use LDAP. OpenID and SAML will cause **irrevocable data loss**. Nextcloud Server-Side Encryption requires access to the user's cleartext password, which is not available through OpenID or SAML.
```

> If you require [Server Side Encryption](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/encryption_configuration.html), you must use LDAP. OpenID and SAML will cause **irrevocable data loss**. Nextcloud Server-Side Encryption requires access to the user's cleartext password, which is not available through OpenID or SAML.

Reviewer 2: ```suggestion
If you require [Server Side Encryption](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/encryption_configuration.html), you must use LDAP. OpenID and SAML will cause **irrevocable data loss**. Nextcloud Server-Side Encryption requires access to the user's cleartext password, which Nextcloud only has access to when using LDAP as the user enters their password directly into Nextcloud.
```

## Authentication

There are 3 ways to setup single sign on (SSO) for Nextcloud:

-   [via OIDC Connect (OAuth)](#openid-connect-auth)
-   [via SAML](#saml-auth)

> -   via LDAP outpost

Reviewer: Since we are adding a third option here, we need to add a section below for it, like the other two. This new sectin could be quite short, just a sentence explaining the process conceptually, and link off to this doc: https://docs.goauthentik.io/docs/add-secure-apps/outposts/#more-about-outpost-integrations.

Or link to the separate docs for K9s vs Docker...

Author: It is not trivial and I do not have a clean configuration to offer, much less a reasonable process. I thought the **data loss** warranted a notice. I am not agreeing to expand the scope of the volunteer work I am doing at this time.

Reviewer: No worries, I can do it later, after we merge your addition! Thanks again for adding the important info.

> -   via LDAP outpost

Reviewer 2: ```suggestion
-   via LDAP outpost (required for SSE, not covered in this documentation)
```
